# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1492863174 14400
#      Sat Apr 22 08:12:54 2017 -0400
# Node ID 58d56f52d50ae8757a21655c6af44a044b4cca15
# Parent  7dfa3bc8e3d53ebf323d3beaf5fc087b8b4633b2
move contents of liboctave/cruft/misc to liboctave/util

* blaswrap.c, cquit.c, d1mach-tst.for, d1mach.f, f77-extern.cc,
f77-fcn.c, f77-fcn.h, i1mach.f, lo-error.c, lo-error.h, quit.cc,
quit.h, r1mach.f: Move from liboctave/cruft/misc to liboctave/util.

* configure.ac, etc/HACKING, libgui/graphics/module.mk,
libgui/src/module.mk, libinterp/module.mk, liboctave/module.mk,
liboctave/cruft/module.mk, liboctave/util/module.mk, src/module.mk:
Update.

* liboctave/cruft/misc/module.mk: Delete.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -992,41 +992,41 @@ fi
 if test $ax_blas_ok = no; then
   case $host_os in
     darwin*)
       ## test if wrapper functions help
       octave_blaswrap_save_CFLAGS="$CFLAGS"
       CFLAGS="$CFLAGS -DUSE_BLASWRAP"
       AC_LANG_PUSH(C)
       AC_COMPILE_IFELSE([AC_LANG_SOURCE([[
-          #include "liboctave/cruft/misc/blaswrap.c"
+          #include "liboctave/util/blaswrap.c"
         ]])],
         [mv conftest.$ac_objext blaswrap.$ac_objext
          octave_blaswrap_save_BLAS_LIBS="$BLAS_LIBS"
          BLAS_LIBS="blaswrap.$ac_objext -framework vecLib"
 
          save_FFLAGS="$FFLAGS"
          FFLAGS="$FFLAGS $F77_INTEGER_8_FLAG"
 
          OCTAVE_BLAS_WITH_F77_FUNC([:], [:])
          AX_LAPACK([:], [:])
 
          ## Restore FFLAGS.
          FFLAGS="$save_FFLAGS"
 
          ## remove temp file
          rm -f blaswrap.$ac_objext],
-        [AC_MSG_FAILURE([cannot compile liboctave/cruft/misc/blaswrap.c])])
+        [AC_MSG_FAILURE([cannot compile liboctave/util/blaswrap.c])])
       AC_LANG_POP(C)
       CFLAGS="$octave_blaswrap_save_CFLAGS"
 
       if test $ax_blas_ok = no; then
         BLAS_LIBS="$octave_blaswrap_save_BLAS_LIBS"
       else
-        ## wrapper in cruft, remove from BLAS_LIBS
+        ## wrapper in liboctave/util, remove from BLAS_LIBS
         BLAS_LIBS=`echo $BLAS_LIBS | $SED -e 's/blaswrap.[[^ ]]* //g'`
         AC_DEFINE(USE_BLASWRAP, 1,
           [Define to 1 if BLAS functions need to be wrapped (potentially needed for 64-bit OSX only).])
       fi
     ;;
   esac
 fi
 
diff --git a/etc/HACKING b/etc/HACKING
--- a/etc/HACKING
+++ b/etc/HACKING
@@ -180,18 +180,16 @@ An overview of the directory layout of O
       dasrt            differential algebraic equation solver with root finding
 
       dassl            differential-algebraic system solver
 
       fftpack          subroutines for fast fourier transforms
 
       lapack-xtra      wrappers for lapack functions used in Octave
 
-      misc             miscellaneous utilities
-
       odepack          ordinary differential equation solver
 
       ordered-qz       code for ordering eigenvalues for QZ factorization
 
       quadpack         subroutines for numerical integration
 
       ranlib           random number generators
 
diff --git a/libgui/graphics/module.mk b/libgui/graphics/module.mk
--- a/libgui/graphics/module.mk
+++ b/libgui/graphics/module.mk
@@ -130,17 +130,16 @@ libgui_graphics_libgui_graphics_la_CPPFL
   $(AM_CPPFLAGS) \
   $(FT2_CPPFLAGS) \
   $(FONTCONFIG_CPPFLAGS) \
   $(HDF5_CPPFLAGS) \
   @OCTGUI_DLL_DEFS@ \
   @QT_CPPFLAGS@ \
   -Ilibgui/graphics -I$(srcdir)/libgui/graphics \
   -Isrc -I$(srcdir)/libgui/src \
-  -I$(srcdir)/liboctave/cruft/misc \
   -I$(srcdir)/liboctave/array \
   -Iliboctave/numeric -I$(srcdir)/liboctave/numeric \
   -Iliboctave/operators -I$(srcdir)/liboctave/operators \
   -I$(srcdir)/liboctave/system \
   -I$(srcdir)/liboctave/util \
   -Ilibinterp -I$(srcdir)/libinterp \
   -Ilibinterp/parse-tree -I$(srcdir)/libinterp/parse-tree \
   -Ilibinterp/corefcn -I$(srcdir)/libinterp/corefcn \
diff --git a/libgui/src/module.mk b/libgui/src/module.mk
--- a/libgui/src/module.mk
+++ b/libgui/src/module.mk
@@ -215,17 +215,16 @@ libgui_src_libgui_src_la_CPPFLAGS = \
   $(FONTCONFIG_CPPFLAGS) \
   @OCTGUI_DLL_DEFS@ \
   @QT_CPPFLAGS@ \
   -I$(srcdir)/libgui/qterminal/libqterminal \
   -Ilibgui/src -I$(srcdir)/libgui/src \
   -I$(srcdir)/libgui/src/m-editor \
   -I$(srcdir)/libgui/src/qtinfo \
   -I$(srcdir)/libgui/graphics \
-  -I$(srcdir)/liboctave/cruft/misc \
   -I$(srcdir)/liboctave/array \
   -Iliboctave/numeric -I$(srcdir)/liboctave/numeric \
   -Iliboctave/operators -I$(srcdir)/liboctave/operators \
   -I$(srcdir)/liboctave/system \
   -I$(srcdir)/liboctave/util \
   -Ilibinterp -I$(srcdir)/libinterp \
   -Ilibinterp/parse-tree -I$(srcdir)/libinterp/parse-tree \
   -Ilibinterp/corefcn -I$(srcdir)/libinterp/corefcn \
diff --git a/libinterp/module.mk b/libinterp/module.mk
--- a/libinterp/module.mk
+++ b/libinterp/module.mk
@@ -3,17 +3,16 @@ libinterp_EXTRA_DIST =
 libinterp_CLEANFILES =
 libinterp_DISTCLEANFILES =
 libinterp_MAINTAINERCLEANFILES =
 
 ## Search local directories before those specified by the user.
 libinterp_liboctinterp_la_CPPFLAGS = \
   @OCTINTERP_DLL_DEFS@ \
   -Iliboctave -I$(srcdir)/liboctave \
-  -I$(srcdir)/liboctave/cruft/misc \
   -I$(srcdir)/liboctave/array \
   -Iliboctave/numeric -I$(srcdir)/liboctave/numeric \
   -Iliboctave/operators -I$(srcdir)/liboctave/operators \
   -I$(srcdir)/liboctave/system \
   -I$(srcdir)/liboctave/util \
   -I$(srcdir)/libinterp/octave-value \
   -Ilibinterp -I$(srcdir)/libinterp \
   -I$(srcdir)/libinterp/operators \
diff --git a/liboctave/cruft/misc/module.mk b/liboctave/cruft/misc/module.mk
deleted file mode 100644
--- a/liboctave/cruft/misc/module.mk
+++ /dev/null
@@ -1,18 +0,0 @@
-CRUFT_SOURCES += \
-  liboctave/cruft/misc/blaswrap.c \
-  liboctave/cruft/misc/cquit.c \
-  liboctave/cruft/misc/d1mach.f \
-  liboctave/cruft/misc/f77-extern.cc \
-  liboctave/cruft/misc/f77-fcn.c \
-  liboctave/cruft/misc/i1mach.f \
-  liboctave/cruft/misc/lo-error.c \
-  liboctave/cruft/misc/quit.cc \
-  liboctave/cruft/misc/r1mach.f
-
-CRUFT_INC += \
-  liboctave/cruft/misc/f77-fcn.h \
-  liboctave/cruft/misc/lo-error.h \
-  liboctave/cruft/misc/quit.h
-
-liboctave_EXTRA_DIST += \
-  liboctave/cruft/misc/d1mach-tst.for
diff --git a/liboctave/cruft/module.mk b/liboctave/cruft/module.mk
--- a/liboctave/cruft/module.mk
+++ b/liboctave/cruft/module.mk
@@ -11,17 +11,16 @@ CRUFT_SOURCES =
 include liboctave/cruft/amos/module.mk
 include liboctave/cruft/blas-xtra/module.mk
 include liboctave/cruft/daspk/module.mk
 include liboctave/cruft/dasrt/module.mk
 include liboctave/cruft/dassl/module.mk
 include liboctave/cruft/Faddeeva/module.mk
 include liboctave/cruft/fftpack/module.mk
 include liboctave/cruft/lapack-xtra/module.mk
-include liboctave/cruft/misc/module.mk
 include liboctave/cruft/odepack/module.mk
 include liboctave/cruft/ordered-qz/module.mk
 include liboctave/cruft/quadpack/module.mk
 include liboctave/cruft/ranlib/module.mk
 include liboctave/cruft/slatec-err/module.mk
 include liboctave/cruft/slatec-fn/module.mk
 
 liboctave/cruft/cruft.def: $(liboctave_cruft_libcruft_la_SOURCES) build-aux/mk-f77-def.sh
diff --git a/liboctave/module.mk b/liboctave/module.mk
--- a/liboctave/module.mk
+++ b/liboctave/module.mk
@@ -6,17 +6,16 @@ liboctave_DISTCLEANFILES =
 liboctave_MAINTAINERCLEANFILES =
 
 ## Search local directories before those specified by the user.
 liboctave_liboctave_la_CPPFLAGS = \
   @OCTAVE_DLL_DEFS@ \
   @CRUFT_DLL_DEFS@ \
   -Iliboctave -I$(srcdir)/liboctave \
   -I$(srcdir)/liboctave/array \
-  -I$(srcdir)/liboctave/cruft/misc \
   -Iliboctave/numeric -I$(srcdir)/liboctave/numeric \
   -Iliboctave/operators -I$(srcdir)/liboctave/operators \
   -I$(srcdir)/liboctave/system \
   -I$(srcdir)/liboctave/util \
   -I$(srcdir)/liboctave/wrappers
 
 liboctave_liboctave_la_CFLAGS = $(AM_CFLAGS) $(WARN_CFLAGS)
 
diff --git a/liboctave/cruft/misc/blaswrap.c b/liboctave/util/blaswrap.c
rename from liboctave/cruft/misc/blaswrap.c
rename to liboctave/util/blaswrap.c
diff --git a/liboctave/cruft/misc/cquit.c b/liboctave/util/cquit.c
rename from liboctave/cruft/misc/cquit.c
rename to liboctave/util/cquit.c
diff --git a/liboctave/cruft/misc/d1mach-tst.for b/liboctave/util/d1mach-tst.for
rename from liboctave/cruft/misc/d1mach-tst.for
rename to liboctave/util/d1mach-tst.for
diff --git a/liboctave/cruft/misc/d1mach.f b/liboctave/util/d1mach.f
rename from liboctave/cruft/misc/d1mach.f
rename to liboctave/util/d1mach.f
diff --git a/liboctave/cruft/misc/f77-extern.cc b/liboctave/util/f77-extern.cc
rename from liboctave/cruft/misc/f77-extern.cc
rename to liboctave/util/f77-extern.cc
--- a/liboctave/cruft/misc/f77-extern.cc
+++ b/liboctave/util/f77-extern.cc
@@ -24,12 +24,12 @@ along with Octave; see the file COPYING.
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
 #include "f77-fcn.h"
 #include "lo-error.h"
 
 // This whole file is a kluge just to avoid unresolved symbol errors
-// when creating shared versions of libcruft.
+// when creating shared versions of liboctave.
 
 // So we can check to see if an exception has occurred.
 int f77_exception_encountered = 0;
diff --git a/liboctave/cruft/misc/f77-fcn.c b/liboctave/util/f77-fcn.c
rename from liboctave/cruft/misc/f77-fcn.c
rename to liboctave/util/f77-fcn.c
diff --git a/liboctave/cruft/misc/f77-fcn.h b/liboctave/util/f77-fcn.h
rename from liboctave/cruft/misc/f77-fcn.h
rename to liboctave/util/f77-fcn.h
diff --git a/liboctave/cruft/misc/i1mach.f b/liboctave/util/i1mach.f
rename from liboctave/cruft/misc/i1mach.f
rename to liboctave/util/i1mach.f
diff --git a/liboctave/cruft/misc/lo-error.c b/liboctave/util/lo-error.c
rename from liboctave/cruft/misc/lo-error.c
rename to liboctave/util/lo-error.c
--- a/liboctave/cruft/misc/lo-error.c
+++ b/liboctave/util/lo-error.c
@@ -26,17 +26,17 @@ along with Octave; see the file COPYING.
 
 #include <stdarg.h>
 #include <stdio.h>
 #include <stdlib.h>
 
 #include "lo-error.h"
 
 /* Having this file in this directory is a kluge to avoid unresolved
-   symbol errors when creating shared versions of libcruft. */
+   symbol errors when creating shared versions of liboctave. */
 
 /* Pointer to the current error handling function. */
 OCTAVE_NORETURN liboctave_error_handler
   current_liboctave_error_handler = liboctave_fatal;
 
 /* Pointer to the current error_with_id handling function. */
 OCTAVE_NORETURN liboctave_error_with_id_handler
   current_liboctave_error_with_id_handler = liboctave_fatal_with_id;
diff --git a/liboctave/cruft/misc/lo-error.h b/liboctave/util/lo-error.h
rename from liboctave/cruft/misc/lo-error.h
rename to liboctave/util/lo-error.h
diff --git a/liboctave/util/module.mk b/liboctave/util/module.mk
--- a/liboctave/util/module.mk
+++ b/liboctave/util/module.mk
@@ -9,82 +9,100 @@ UTIL_INC = \
   liboctave/util/functor.h \
   liboctave/util/glob-match.h \
   liboctave/util/lo-array-errwarn.h \
   liboctave/util/lo-array-gripes.h \
   liboctave/util/lo-cutils.h \
   liboctave/util/lo-hash.h \
   liboctave/util/lo-ieee.h \
   liboctave/util/lo-math.h \
+  liboctave/util/lo-regexp.h \
   liboctave/util/lo-traits.h \
   liboctave/util/lo-utils.h \
+  liboctave/util/f77-fcn.h \
+  liboctave/util/lo-error.h \
+  liboctave/util/quit.h \
   liboctave/util/oct-base64.h \
   liboctave/util/oct-binmap.h \
   liboctave/util/oct-cmplx.h \
   liboctave/util/oct-glob.h \
+  liboctave/util/oct-inttypes-fwd.h \
   liboctave/util/oct-inttypes.h \
-  liboctave/util/oct-inttypes-fwd.h \
   liboctave/util/oct-locbuf.h \
   liboctave/util/oct-mutex.h \
   liboctave/util/oct-refcount.h \
   liboctave/util/oct-rl-edit.h \
   liboctave/util/oct-rl-hist.h \
-  liboctave/util/oct-string.h \
   liboctave/util/oct-shlib.h \
   liboctave/util/oct-sort.h \
+  liboctave/util/oct-string.h \
   liboctave/util/pathsearch.h \
-  liboctave/util/lo-regexp.h \
   liboctave/util/singleton-cleanup.h \
   liboctave/util/sparse-sort.h \
   liboctave/util/sparse-util.h \
   liboctave/util/str-vec.h \
   liboctave/util/sun-utils.h \
   liboctave/util/unwind-prot.h \
   liboctave/util/url-transfer.h
 
 NOINSTALL_UTIL_INC = \
   liboctave/util/kpse.h \
   liboctave/util/oct-sparse.h
 
+UTIL_F77_SRC = \
+  liboctave/util/d1mach.f \
+  liboctave/util/i1mach.f \
+  liboctave/util/r1mach.f
+
 UTIL_C_SRC = \
   liboctave/util/f2c-main.c \
   liboctave/util/lo-cutils.c \
+  liboctave/util/blaswrap.c \
+  liboctave/util/cquit.c \
+  liboctave/util/f77-fcn.c \
+  liboctave/util/lo-error.c \
   liboctave/util/oct-rl-edit.c \
   liboctave/util/oct-rl-hist.c
 
 UTIL_SRC = \
   liboctave/util/cmd-edit.cc \
   liboctave/util/cmd-hist.cc \
   liboctave/util/data-conv.cc \
   liboctave/util/glob-match.cc \
   liboctave/util/kpse.cc \
   liboctave/util/lo-array-errwarn.cc \
   liboctave/util/lo-array-gripes.cc \
   liboctave/util/lo-hash.cc \
   liboctave/util/lo-ieee.cc \
+  liboctave/util/lo-regexp.cc \
   liboctave/util/lo-utils.cc \
+  liboctave/util/f77-extern.cc \
+  liboctave/util/quit.cc \
   liboctave/util/oct-base64.cc \
   liboctave/util/oct-glob.cc \
   liboctave/util/oct-inttypes.cc \
   liboctave/util/oct-locbuf.cc \
   liboctave/util/oct-mutex.cc \
-  liboctave/util/oct-string.cc \
   liboctave/util/oct-shlib.cc \
   liboctave/util/oct-sparse.cc \
+  liboctave/util/oct-string.cc \
   liboctave/util/pathsearch.cc \
-  liboctave/util/lo-regexp.cc \
   liboctave/util/singleton-cleanup.cc \
   liboctave/util/sparse-sort.cc \
   liboctave/util/sparse-util.cc \
   liboctave/util/str-vec.cc \
   liboctave/util/unwind-prot.cc \
   liboctave/util/url-transfer.cc \
+  $(UTIL_F77_SRC) \
   $(UTIL_C_SRC) \
   $(NOINSTALL_UTIL_INC)
 
+liboctave_EXTRA_DIST += \
+  liboctave/util/d1mach-tst.for
+
 LIBOCTAVE_TEMPLATE_SRC += \
   liboctave/util/oct-sort.cc
 
 noinst_LTLIBRARIES += liboctave/util/libutil.la
 
 liboctave_util_libutil_la_SOURCES = $(UTIL_SRC)
 
 liboctave_util_libutil_la_CPPFLAGS = \
diff --git a/liboctave/cruft/misc/quit.cc b/liboctave/util/quit.cc
rename from liboctave/cruft/misc/quit.cc
rename to liboctave/util/quit.cc
diff --git a/liboctave/cruft/misc/quit.h b/liboctave/util/quit.h
rename from liboctave/cruft/misc/quit.h
rename to liboctave/util/quit.h
diff --git a/liboctave/cruft/misc/r1mach.f b/liboctave/util/r1mach.f
rename from liboctave/cruft/misc/r1mach.f
rename to liboctave/util/r1mach.f
diff --git a/src/module.mk b/src/module.mk
--- a/src/module.mk
+++ b/src/module.mk
@@ -4,17 +4,16 @@ src_CLEANFILES =
 src_DISTCLEANFILES =
 src_MAINTAINERCLEANFILES =
 
 ## Search local directories before those specified by the user.
 
 SRC_DIR_CPPFLAGS = \
   -Iliboctave -I$(srcdir)/liboctave \
   -I$(srcdir)/liboctave/array \
-  -I$(srcdir)/liboctave/cruft/misc \
   -I$(srcdir)/liboctave/numeric \
   -I$(srcdir)/liboctave/system \
   -I$(srcdir)/liboctave/util \
   -Iliboctave/wrappers -I$(srcdir)/liboctave/wrappers \
   -Ilibinterp -I$(srcdir)/libinterp \
   -Ilibinterp/corefcn -I$(srcdir)/libinterp/corefcn \
   -I$(srcdir)/src
 
